joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h

```diff
@@ -193,11 +193,16 @@ checkPathTolerances(const typename Segment::State& state_error,
     rt_segment_goal->setAborted(rt_segment_goal->preallocated_result_);
     rt_active_goal_.reset();
     
+    // Sleep until the goal is aborted
+    ros::Duration action_monitor_period_2x=action_monitor_period_.operator *(2);
+    action_monitor_period_2x.sleep();
```

Comment:
sleeping in the RT update cycle is a no-go.

Reply:
ok, understand. It turns out that the [update](https://github.com/ros-controls/ros_controllers/blob/indigo-devel/joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h#L374) loop after calling [setHoldPosition](https://github.com/ros-controls/ros_controllers/blob/indigo-devel/joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h#L671) can destroy the [goal_handle_timer_](https://github.com/ros-controls/ros_controllers/blob/indigo-devel/joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h#L567). So I make update-thread sleep for awhile until the timer run at least one more time for aborting the action. There should be many solutions without sleeping.